### PR TITLE
Lehdari rendering

### PR DESCRIPTION
**Pull request by Lehdari**
GUI Rendering pipeline

### CODE REVIEW

File: src/gui/Mesh.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    glGenVertexArrays(1, &vertexArrayObjectId_);
+    glBindVertexArray(vertexArrayObjectId_);
+
+    //  upload the vertex data to GPU and set up the vertex attribute arrays
+    glGenBuffers(1, &positionBufferId_);
+    glBindBuffer(GL_ARRAY_BUFFER, positionBufferId_);
+    glBufferData(GL_ARRAY_BUFFER, positions.size() * sizeof(std::array<float, 4>), &positions[0], GL_STATIC_DRAW);
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, 0, (GLvoid*)0);
+/*
```
Comment (tkln) [inline src/gui/Mesh.cpp]: Should this be enabled or removed (probably with rest of the texture coordinate parsing)?

File: src/gui/Mesh.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+void Mesh::render(const Shader& shader, const Camera& camera, const Vec3GLf& color) const {
+    shader.useShader(camera.getVP() * getOrientation(), color);
+    glBindVertexArray(vertexArrayObjectId_);
+
+    glDrawElements(GL_TRIANGLES, nIndices_, GL_UNSIGNED_INT, (GLvoid*)0);
+
+    glBindVertexArray(0);
+}
+#if 0
```
Comment (tkln) [inline src/gui/Mesh.cpp]: This probably should be just removed.

File: include/gui/Mesh.hpp [hunk 3]
```diff
@@ -27,12 +27,10 @@ namespace gui {
...
         void render(const Shader& shader,
                     const Camera& camera,
+                    const Mat4GLf& orientation,
                     const Vec3GLf& color = Vec3GLf(1.0f, 1.0f, 1.0f)) const;
 
-        void setPosition(const Vec3GLf& position);
-        void setRotation(const Mat3GLf& rotation);
-
-        Mat4GLf getOrientation(void) const;
+        //Mat4GLf getOrientation(void) const;
```
Comment (tkln) [inline include/gui/Mesh.hpp]: The title says remove, but these are just commented out.

File: src/gui/Shader.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+GLuint Shader::getId(void) const {
+    return programId_;
+}
+
+void Shader::useShader(const Mat4GLf& mvp, const Vec3GLf& color) const {
+
+
+    /*glBindFragDataLocation(programId_, 0, "pos");
```
Comment (tkln) [inline src/gui/Shader.cpp]: I think these should be removed.

File: src/app/SDLApp.cpp [hunk 5]
```diff
@@ -110,6 +129,7 @@ void SDLApp::handleEvents(SDL_Event& event)
 
 void SDLApp::render(void)
 {
+    /*
```
Comment (tkln) [inline src/app/SDLApp.cpp]: Please don't break imgui

File: src/gui/GuiPuma560.cpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+void GuiPuma560::loadLinkMeshes()
+{
+    _links[0].loadFromObj("../res/models/puma_link1_mod.obj");
+    _links[1].loadFromObj("../res/models/puma_link2_mod.obj");
+    _links[2].loadFromObj("../res/models/puma_link3_mod.obj");
+    _links[3].loadFromObj("../res/models/puma_link4_mod.obj");
+    _links[4].loadFromObj("../res/models/puma_link5_mod.obj");
+    _links[5].loadFromObj("../res/models/puma_link6_mod.obj");
```
Comment (sndels) [inline src/gui/GuiPuma560.cpp]: Ditto, paths

**Comment by sndels** [comment]
Made the requested changes to original commits as well as a few additional ones that made sense to split to their own.

**Pull request merged**